src/Core/OrganizationFeatures/OrganizationServiceCollectionExtensions.cs

```diff
@@ -47,6 +49,9 @@ public static void AddOrganizationServices(this IServiceCollection services, IGl
         services.AddOrganizationAuthCommands();
         services.AddOrganizationUserCommandsQueries();
         services.AddBaseOrganizationSubscriptionCommandsQueries();
+
+        services.AddScoped<IAuthorizationHandler, CollectionUserAuthorizationHandler>();
+        services.AddScoped<IAuthorizationHandler, CollectionGroupAuthorizationHandler>();
```

Comment:
This class is used by many different projects (e.g. scim, billing, identity, etc) but these dependencies are only needed for Api. I suggest registering them in a serviceCollectionExtension class that only Api uses. (maybe a dedicated `AuthorizationHandlerServiceCollectionExtension`?)

That suggests to me that these should also be moved to somewhere in the Api project, rather than in Core.

Reply:
I agree that it shouldn't be added in that shared organization service collection extension method, as its called in places this handler is not needed. 

However, I still feel that they should belong in the `Core` project with its own `AddAuthorizationHandlers()` service collection extension. This is also stays consistent with what SecretsManager did with their authorization handlers and registration. Authorization, as a general concept, is something that spans across many projects in some form or another, so to me, it makes sense to keep these together and at a higher level than the API project. 

If you feel strongly that it should be moved into the API project let me know and we can discuss!